packages/plugin-ext/src/hosted/browser/worker/worker-main.ts

```diff
@@ -59,9 +60,22 @@ rpc.set(MAIN_RPC_CONTEXT.HOSTED_PLUGIN_MANAGER_EXT, new HostedPluginManagerExtIm
     },
     stopPlugins(contextPath: string, pluginIds: string[]): void {
         pluginIds.forEach(pluginId => {
-            const stopPluginMethod = plugins.get(pluginId);
-            if (stopPluginMethod) {
-                stopPluginMethod();
+            const pluginData = plugins.get(pluginId);
+            if (pluginData) {
+                // dispose any objects
+                const pluginContext = pluginData.pluginContext;
+                if (pluginContext) {
+                    pluginContext.subscriptions.forEach((element: Disposable) => {
+                        element.dispose();
+                    });
+                }
+
+                // call stop method
+                if (pluginData.stopPluginMethod) {
+                    pluginData.stopPluginMethod();
```

Comment:
Shouldn't we call this before disposing?

Reply:
Yes, agree.
Fixed in both places